app/templates/play/game-dev-goals.jade

```diff
@@ -22,3 +22,5 @@ mixin game-dev-goals(goals)
           li.goal(data-i18n="play_game_dev_level.goal_collect_amount", data-i18n-options={amount: goal.config.amount})
         else
           li.goal(data-i18n="play_game_dev_level.goal_collect")
+      else if goal.type == "user_defined"
+        li.goal= goal.directions
```

Comment:
What is the relationship between `@world.uiText.directions` and `goal.directions`?

Our naming is a little confusing across code areas here.  E.g. why don't we set some sort of goal property on the world object instead of setting UI text?